File: SDKs/velopayments.com/2.26.127/python/src/sdk/models/operations/getwebhookv1.py
from dataclasses import dataclass, field
from typing import Any,Optional
from sdk.models import shared


@dataclass
class GetWebhookV1PathParams:
    webhook_id: str = field(metadata={'path_param': { 'field_name': 'webhookId', 'style': 'simple', 'explode': False }})
    

@dataclass
class GetWebhookV1Request:
    path_params: GetWebhookV1PathParams = field()
    

@dataclass
class GetWebhookV1Response:
    content_type: str = field()
    status_code: int = field()
    webhook_response: Optional[shared.WebhookResponse] = field(default=None)
    inline_response_401: Optional[Any] = field(default=None)
    inline_response_403: Optional[Any] = field(default=None)
    inline_response_404: Optional[Any] = field(default=None)
    

File: SDKs/velopayments.com/2.26.127/typescript/src/sdk/models/operations/getwebhookv1.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetWebhookV1PathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=webhookId" })
  webhookId: string;
}


export class GetWebhookV1Request extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetWebhookV1PathParams;
}


export class GetWebhookV1Response extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  webhookResponse?: shared.WebhookResponse;

  @SpeakeasyMetadata()
  inlineResponse401?: any;

  @SpeakeasyMetadata()
  inlineResponse403?: any;

  @SpeakeasyMetadata()
  inlineResponse404?: any;
}
